Docs: Add security warning for `--host 0.0.0.0` in CLI options (#35241)
storybookjs/storybook: docs/api/cli-options.mdx

| `-V`, `--version`               | Output the version number.<br />`storybook dev -V`                                                                                                                                                                                                  |
| `-p`, `--port [number]`         | Port to run Storybook.<br />`storybook dev -p 9009`                                                                                                                                                                                                 |
| `--exact-port`                  | Attempts to run Storybook on the exact port number specified.<br />If the port is already in use, Storybook will exit with an error message.<br />`storybook dev -p 9009 --exact-port`                                                              |

> | `-h`, `--host [string]`         | Host to run Storybook. Setting this to `0.0.0.0` binds to all network interfaces. When doing so, consider also configuring [`core.allowedHosts`](./main-config/main-config-core.mdx#allowedhosts) to restrict which hosts can access your Storybook instance.<br />`storybook dev -h my-host.com`                                                                                                                                                                                          |

Reviewer: @ghengeveld when you're able, adjust this to the following:
```suggestion
| `-h`, `--host [string]`         | Host to run Storybook. Setting this to `0.0.0.0` binds to all network interfaces. <br/>Extend it with [`core.allowedHosts`](./main-config/main-config-core.mdx#allowedhosts) to restrict which hosts can access your Storybook instance.<br />`storybook dev -h my-host.com` |
```

Also, can you check what's happening with the table? It seems a bit unformatted.

Options include:

| Option                   | Description |
| ------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
| `-h`, `--help`           | Output usage information.<br />`storybook remove --help` |

> | `--package-manager`      | Sets the package manager to use when removing the addon.<br />Available package managers include `npm`, `yarn`, and `pnpm`.<br />`storybook remove [addon]--package-manager pnpm` |

Reviewer 2: @copilot fix please
